app/versiondb_placeholder.go

```diff
@@ -0,0 +1,25 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+//go:build !rocksdb
+// +build !rocksdb
+
+package app
+
+import (
+	"errors"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func setupVersionDB(
```

Comment:
versionDB can only be used when building the binary with rocksDB. In such case, the binary compiles with the code in `versiondb.go` file. Otherwise, if compiled with other DB Backend, will compile with this code.

Probably we can refactor this function to take the DBBackend as input arg instead of having two files for each of the builds

Reply:
Actually this would not work, cause will require to be compiled with rocksdb.
When trying to do so, go the errors:

```
# github.com/linxGnu/grocksdb
../../go/pkg/mod/github.com/linx!gnu/grocksdb@v1.8.4/cache.go:25:12: could not determine kind of name for C.rocksdb_cache_create_hyper_clock
../../go/pkg/mod/github.com/linx!gnu/grocksdb@v1.8.4/cache.go:31:12: could not determine kind of name for C.rocksdb_cache_create_hyper_clock_opts
../../go/pkg/mod/github.com/linx!gnu/grocksdb@v1.8.4/cache.go:19:12: could not determine kind of name for C.rocksdb_cache_create_lru_opts
../../go/pkg/mod/github.com/linx!gnu/grocksdb@v1.8.4/cache.go:82:2: could not determine kind of name for C.rocksdb_cache_disown_data
...
```

Reply:
yeah, the binary need to build with librocksdb, so it has to be a compile time check.